test/suite/test_gc03.py

```diff
@@ -0,0 +1,156 @@
+        uri = "table:gc03"
+        ds = SimpleDataSet(
+            self, uri, 0, key_format="i", value_format="S", config='log=(enabled=false)')
+        ds.populate()
+
+        # Pin oldest and stable to timestamp 1.
+        self.conn.set_timestamp('oldest_timestamp=' + timestamp_str(1) +
+            ',stable_timestamp=' + timestamp_str(1))
+
+        bigvalue = "aaaaa" * 100
+        bigvalue2 = "ddddd" * 100
+        self.large_updates(uri, bigvalue, ds, nrows, 10)
+
+        # Check that all updates are seen
+        #self.check(bigvalue, uri, nrows, 20)
+
+        self.large_updates(uri, bigvalue2, ds, nrows, 20)
+
+        # Check that the new updates are only seen after the update timestamp
+        #self.check(bigvalue2, uri, nrows, 100)
+
+        # Pin oldest and stable to timestamp 100.
+        self.conn.set_timestamp('oldest_timestamp=' + timestamp_str(100) +
+            ',stable_timestamp=' + timestamp_str(100))
+
+        # Checkpoint to ensure that the history store is gets populated and cleaned
+        self.session.checkpoint()
+
+        # Check that the new updates are only seen after the update timestamp
+        #self.check(bigvalue, uri, nrows, 100)
+
+        # Load a slight modification with a later timestamp.
+        self.large_modifies(uri, 'A', ds, 10, 1, nrows, 110)
+        self.large_modifies(uri, 'B', ds, 20, 1, nrows, 120)
+        self.large_modifies(uri, 'C', ds, 30, 1, nrows, 130)
+
+        # Second set of update operations with increased timestamp
+        self.large_updates(uri, bigvalue2, ds, nrows, 200)
+
+        # Check that the new updates are only seen after the update timestamp
+        #self.check(bigvalue2, uri, nrows, 200)
+
+        # Pin oldest and stable to timestamp 300.
+        self.conn.set_timestamp('oldest_timestamp=' + timestamp_str(200) +
+            ',stable_timestamp=' + timestamp_str(200))
+
+        # Checkpoint to ensure that the history store is gets populated and cleaned
+        self.session.checkpoint()
+
+        # Check that the new updates are only seen after the update timestamp
+        #self.check(bigvalue2, uri, nrows, 200)
+
+        # Load a slight modification with a later timestamp.
+        self.large_modifies(uri, 'A', ds, 10, 1, nrows, 210)
+        self.large_modifies(uri, 'B', ds, 20, 1, nrows, 220)
+        self.large_modifies(uri, 'C', ds, 30, 1, nrows, 230)
+
+        # Third set of update operations with increased timestamp
+        self.large_updates(uri, bigvalue, ds, nrows, 300)
+
+        # Check that the new updates are only seen after the update timestamp
+        #self.check(bigvalue, uri, nrows, 300)
+
+        # Pin oldest and stable to timestamp 400.
+        self.conn.set_timestamp('oldest_timestamp=' + timestamp_str(300) +
+            ',stable_timestamp=' + timestamp_str(300))
+
+        # Checkpoint to ensure that the history store is gets populated and cleaned
+        self.session.checkpoint()
```

Comment:
Are there, or will there be, any statistics we can check here to make sure that this happened?

Reply:
The stats ticket is not yet implemented, Once all the implementation tickets are finished, we have a test ticket that should take care of all the pending test changes.